### PR TITLE
pull request to merge 1709 changes

### PR DESCRIPTION
**Pull request by soumyanishan**

### CODE REVIEW

File: VMBackup/main/handle.sh [hunk 1]
```diff
@@ -0,0 +1,38 @@
+#!/bin/bash
+##/var/log/azure/Microsoft.Azure.RecoveryServices.VMSnapshotLinux/
+pwdcommand=`pwd`
+pwdstr="$pwdcommand"
+logfile='/var/log/azure/Microsoft.Azure.RecoveryServices.VMSnapshotLinux/'${pwdstr#*-}'/shell.log'
```
Comment (vityagi) [inline VMBackup/main/handle.sh]: Get this path from HandlerEnvironment.json

File: VMBackup/main/handle.sh [hunk 2]
```diff
@@ -0,0 +1,38 @@
...
+if [ "$1" == "enable" ]; then
+    echo "`date`- kill existing process if exists" >> $logfile
+    kill $(ps aux | grep 'main/handle.py' | awk '{print $2}')
+fi
+if [ -f "/usr/bin/python2" ]
+then
+    echo "`date`- python 2 path exists" >> $logfile
+    /usr/bin/python2 main/handle.py -$1
+    rc=$?
+    if [[ $rc != 0 ]]
```
Comment (vityagi) [inline VMBackup/main/handle.sh]: check whether if-else is needed

File: VMBackup/main/Utils/HandlerUtil.py [hunk 3]
```diff
@@ -492,14 +493,26 @@ def do_status_report(self, operation, status, status_code, message, taskId = Non
...
-        return stat_rept
+
+    def is_status_file_exists(self):
+        try:
+            if os.path.exists(self._context._status_file):
+                return True
+            else:
+                return False
+        except Exception as e:
+            self.log("exception is getting status file")
```
Comment (vityagi) [inline VMBackup/main/Utils/HandlerUtil.py]: add stacktrace

File: VMBackup/main/Utils/HandlerUtil.py [hunk 4]
```diff
@@ -562,3 +575,12 @@ def get_prev_log(self):
             return ''.join(str(x) for x in lines)
         else:
             return ''.join(str(x) for x in lines)
+    
+    def get_shell_script_log(self):
+        with open(self._context._shell_log_file, "r") as f:
+            lines = f.readlines()
+        if(len(lines) > 10):
+            lines = lines[-10:]
+            return ''.join(str(x) for x in lines)
```
Comment (vityagi) [inline VMBackup/main/Utils/HandlerUtil.py]: move return after if-else, no need of else

File: VMBackup/main/Utils/HandlerUtil.py [hunk 5]
```diff
@@ -562,3 +575,12 @@ def get_prev_log(self):
             return ''.join(str(x) for x in lines)
         else:
             return ''.join(str(x) for x in lines)
+    
+    def get_shell_script_log(self):
+        with open(self._context._shell_log_file, "r") as f:
+            lines = f.readlines()
```
Comment (vityagi) [inline VMBackup/main/Utils/HandlerUtil.py]: declare lines outside before open

File: VMBackup/main/handle.py [hunk 6]
```diff
@@ -100,9 +100,30 @@ def timedelta_total_seconds(delta):
     else:
         return delta.total_seconds()
 
-def status_report(status, status_code, message, snapshot_info = None):
+def status_report_to_file(file_report_msg):
+    global backup_logger,hutil
+    hutil.write_to_status_file(file_report_msg)
+    backup_logger.log(file_report_msg,True)
```
Comment (vityagi) [inline VMBackup/main/handle.py]: Add text "file_report_msg" before status

File: VMBackup/main/handle.py [hunk 7]
```diff
@@ -100,9 +100,30 @@ def timedelta_total_seconds(delta):
...
+    try:
+        if(para_parser is not None and para_parser.statusBlobUri is not None and para_parser.statusBlobUri != ""):
+            blobWriter = BlobWriter(hutil)
+            if(blob_report_msg is not None):
+                blobWriter.WriteBlob(blob_report_msg,para_parser.statusBlobUri)
+                backup_logger.log("blob status report message:",True)
+                backup_logger.log(blob_report_msg,True)
+            else:
+                backup_logger.log("blob_report_msg is none",True)
+    except Exception as e:
```
Comment (vityagi) [inline VMBackup/main/handle.py]: Log stacktrace

File: VMBackup/main/handle.py [hunk 8]
```diff
@@ -543,16 +595,23 @@ def enable():
             taskIdentity = TaskIdentity()
             taskIdentity.save_identity(para_parser.taskId)
         hutil.save_seq()
-        status_upload_thread=Thread(target=thread_for_status_upload)
-        status_upload_thread.start() 
+        temp_status= 'transitioning'
+        temp_result=CommonVariables.success
+        temp_msg='Transitioning state in enable'
+        blob_report_msg, file_report_msg = get_status_to_report(temp_status, temp_result, temp_msg, None)
+        file_status_upload_thread=Thread(target=status_report_to_file, args=(file_report_msg,))
```
Comment (vityagi) [inline VMBackup/main/handle.py]: Test with python 2.6

File: VMBackup/main/handle.py [hunk 9]
```diff
@@ -543,16 +595,23 @@ def enable():
...
+        file_status_upload_thread.start()
+        blob_status_upload_thread=Thread(target=status_report_to_blob, args=(blob_report_msg,))
+        blob_status_upload_thread.start()
```
Comment (vityagi) [inline VMBackup/main/handle.py]: Validate content of status in status-file and blob

File: VMBackup/main/handle.py [hunk 10]
```diff
@@ -172,31 +185,58 @@ def exit_if_same_taskId(taskId):
...
+                    config.remove_option('SnapshotThread', key)
+                    config.set('SnapshotThread', key, value)
+                else:
+                    config.set('SnapshotThread', key, value)
+            else:
+                config.add_section('SnapshotThread')
+                config.set('SnapshotThread', key, value)
+        else:
+            config.add_section('SnapshotThread')
+            config.set('SnapshotThread', key, value)
```
Comment (vityagi) [inline VMBackup/main/handle.py]: config.set('SnapshotThread', key, value)  is common for all if-elses, can be moved outside after if-elses

File: VMBackup/main/handle.sh (hunk 2 above)
Comment (soumyanishan) [reply VMBackup/main/handle.sh]: done

File: VMBackup/main/Utils/HandlerUtil.py (hunk 3 above)
Comment (soumyanishan) [reply VMBackup/main/Utils/HandlerUtil.py]: done

File: VMBackup/main/Utils/HandlerUtil.py (hunk 4 above)
Comment (soumyanishan) [reply VMBackup/main/Utils/HandlerUtil.py]: done

File: VMBackup/main/handle.sh (hunk 1 above)
Comment (soumyanishan) [reply VMBackup/main/handle.sh]: done

File: VMBackup/main/Utils/HandlerUtil.py (hunk 5 above)
Comment (soumyanishan) [reply VMBackup/main/Utils/HandlerUtil.py]: done

File: VMBackup/main/handle.py (hunk 6 above)
Comment (soumyanishan) [reply VMBackup/main/handle.py]: done

File: VMBackup/main/handle.py (hunk 7 above)
Comment (soumyanishan) [reply VMBackup/main/handle.py]: done

File: VMBackup/main/handle.py (hunk 10 above)
Comment (soumyanishan) [reply VMBackup/main/handle.py]: done

**Pull request merged**